resource.cpp

```diff
@@ -1957,3 +1957,18 @@ bool isValidRConfigGroup(const QString &str)
 
     return result == groupList.size();
 }
+
+/*! Helper function to expose \p resource item as a different type through REST API than internally defined. */
+QVariant R_ItemToRestApiVariant(const ResourceItem *item)
+{
+    if (item)
+    {
+        const ResourceItemDescriptor rid = item->descriptor();
+        if (rid.suffix == RAttrNwkAddress)
+        {
+            return QString("0x") + QString("%1").arg(item->toNumber(), 4, 16, QLatin1Char('0')).toUpper();
```

Comment:
I suggest a small change to save two QString constructions :)

```
return QString("0x%1").arg(item->toNumber(), 4, 16, QLatin1Char('0')).toUpper();
```

Reply:
I'm afraid, I did that on purpose, as the suggested amendment does not keep the small letter X. See the output of https://github.com/dresden-elektronik/deconz-rest-plugin/blob/04bd307a62abf2ef84235b2069a666c03c6d401f/de_web_plugin.cpp#L4382-L4383 in the respective logs.